js/ui/overlay/ui.overlay.js

```diff
@@ -986,18 +979,16 @@ const Overlay = Widget.inherit({
             return;
         }
 
-        const startEventName = addNamespace(dragEventStart, this.NAME);
-        const updateEventName = addNamespace(dragEventMove, this.NAME);
-
-        eventsEngine.off($dragTarget, startEventName);
-        eventsEngine.off($dragTarget, updateEventName);
-
-        if(!this.option('dragEnabled')) {
-            return;
-        }
+        const updatePositionChangeHandled = (value) => this._positionChangeHandled = value;
 
-        eventsEngine.on($dragTarget, startEventName, (e) => { this._dragStartHandler(e); });
-        eventsEngine.on($dragTarget, updateEventName, (e) => { this._dragUpdateHandler(e); });
+        this._drag = new OverlayDrag({
```

Comment:
What do you think about interacting with an existing class instance? 
Update events, container[, etc]. instead of creating a new instance?

```js
const config = {
  dragEnabled: this.option('dragEnabled'),
  handle: $dragTarget.get(0),
  container: this._getDragResizeContainer().get(0),
  draggableElement: this._$content.get(0),
  outsideMultiplayer: this.option('dragArea')?.outsideMultiplayer ?? 0,
  updatePositionChangeHandled
};

if(this._drag) {
  this._drag.init(config);
} else {
  this._drag = new OverlayDrag(config);
}
```

Reply:
seem good, implemented this